src/main/java/fi/okm/jod/yksilo/service/profiili/YksilonOsaaminenService.java

```diff
@@ -96,7 +109,13 @@ public YksilonOsaaminenDto get(JodUser user, UUID id) {
         .orElseThrow(() -> new NotFoundException("Not found"));
   }
 
-  public void delete(JodUser user, UUID id) {
-    repository.deleteByYksiloIdAndId(user.getId(), id);
+  public void delete(JodUser user, Set<UUID> ids) {
+    if (repository.deleteByYksiloIdAndIdIn(user.getId(), ids) != ids.size()) {
+      throw new NotFoundException("Not found");
+    }
+  }
+
+  void deleteAll(Set<YksilonOsaaminen> osaamiset) {
+    repository.deleteAll(osaamiset);
```

Comment:
delete all deletes enties one by one. Consider using deleteAllInBatch, which generates only single query.

Reply:
For performance, that is a good suggestion but bypassing the persistence context can be foot-gun as well...in this case, perhaps the optimization could be applied.